crates/autopilot/src/decoded_settlement.rs

```diff
@@ -265,47 +251,56 @@ impl DecodedSettlement {
         })
     }
 
-    /// Returns unsubsidized fees for all trades.
-    /// Length of the returned vector is equal to the length of `self.trades`
-    /// and `order_fees`.
+    /// Returns fees for all trades.
     pub fn all_fees(
         &self,
         external_prices: &ExternalPrices,
-        order_fees: &[(OrderUid, Option<U256>)],
+        domain_separator: &DomainSeparator,
     ) -> Vec<Fees> {
         self.trades
             .iter()
-            .zip(order_fees.iter())
-            .map(|(trade, (order, order_fee))| {
-                self.fee(external_prices, *order, *order_fee, trade)
-                    .unwrap_or_else(|| {
-                        tracing::warn!("possible incomplete fee calculation");
-                        // we should have an order execution for every trade
-                        Fees {
-                            order: *order,
-                            sell: U256::zero(),
-                            native: U256::zero(),
-                        }
-                    })
+            .filter_map(|trade| {
+                let order = match trade.uid(domain_separator, &self.tokens) {
```

Comment:
The only way `.uid()` could fail if we are not able to recover the owner from the signature. But since this is already verified by the smart contract I think we can simply do `.expect("enforced by smart contract")` to make the error handling a bit leaner.

Same argument applies to `self.fee()` but there are lot more calls that would need to be `.expect()`-ed so I think there it might be not the best idea.

Reply:
> The only way .uid() could fail if we are not able to recover the owner from the signature

Hm, theoretically also depends on the supporting code, for example, how we populated the `OrderData` struct, and this can have bugs (if not now then potentially in the future).

Reply:
Ahh, you're right. :+1: